polars/polars-arrow/src/compute/arithmetics/decimal/div.rs

```diff
@@ -0,0 +1,43 @@
+use super::*;
+
+#[inline]
+fn decimal_div(a: i128, b: i128, scale: i128) -> i128 {
+    // The division is done using the numbers without scale.
+    // The dividend is scaled up to maintain precision after the
+    // division
```

Comment:
The `a * scale` will overflow very often if dealing with high-precision decimals. Basically what you're doing here implies that using scale greater than `MAX_SCALE / 2` is not possible and will lead to overflow errors, I think.

Usually it's done in the 256-bit space (IIRC that's how arrow2 does it too).